fast64_internal/oot/oot_utility.py

```diff
@@ -602,3 +633,282 @@ def oot_utility_register():
 def oot_utility_unregister():
 	for cls in reversed(oot_utility_classes):
 		unregister_class(cls)
+
+def getActorParameter(object, field, shift, mask):
+	'''Returns the actor parameter with the shifts'''
+	attr = getattr(object, field, '0x00')
+	if attr != '0x00' and attr is not None:					
+		if isinstance(attr, str):
+			if shift != '0':
+				return f"(({attr} << {shift}) & {mask})"
+			else:
+				return f"({attr} & {mask})" 
+		elif isinstance(attr, bool):
+			if attr:
+				return f"(1 << {shift})"
+			else:
+				return ""
+		else:
+			raise NotImplementedError
+	else:
+		return ""
+
+def getActorFinalParameters(detailedProp, actorKey, paramTarget, field):
+	'''Returns the full parameter string'''
+	params = ""
+	if field is not None:
+		panelParams = getattr(detailedProp, field, "")
+		if panelParams == "":
+			return "0x0"
+	for actorNode in root:
+		if actorKey == actorNode.get('Key'):
+			lenProp = getMaxElemIndex(actorKey, 'Property', None)
+			lenSwitch = getMaxElemIndex(actorKey, 'Flag', 'Switch')
+			lenBool = getMaxElemIndex(actorKey, 'Bool', None)
+			lenEnum = getMaxElemIndex(actorKey, 'Enum', None)
+			for elem in actorNode:
+				actorType = getattr(detailedProp, actorKey + '.type', None)
+				if hasTiedParams(elem.get('TiedParam'), actorType):
+					params += getActorString(elem, detailedProp, actorKey, lenProp, lenSwitch, lenBool, lenEnum, paramTarget)
+	params = params.rstrip().rstrip('|').rstrip()
+	if paramTarget == 'Params':
+		actorType = getattr(detailedProp, actorKey + '.type', '0')
+		if params == '':
+			return f'0x{actorType}'
+		else:
+			return f'(0x{actorType} | ({params}))'
+	elif params != '':
+		return params
+	else:
+		return '0x0'
+
+def setActorParameter(object, field, param, mask):
+	'''Sets the attributes to have the correct display on the UI'''
+	shift = len(f'{mask:016b}') - len(f'{mask:016b}'.rstrip('0'))
+	if field.endswith('.type'):
+		setattr(object, field, f'{param & mask:04X}')
+	else:
+		attr = getattr(object, field, '0x0')
+		if isinstance(attr, str):
+			setattr(object, field, f'0x{(param & mask) >> shift:02X}')
+		elif isinstance(attr, bool):
+			setattr(object, field, bool((param & mask) >> shift))
+		else:
+			raise NotImplementedError
+
+def getMaxElemIndex(actorKey, elemTag, flagType):
+	'''Looking for the highest index for the current actor'''
```

Comment:
I'm not sure where the "highest" part of this is implemented in this function? there's no max() or comparison

Reply:
it returns the last index string from the XML, I should reword it this way: "Looking for the last index of the current actor"